### PR TITLE
Switch streaming encoding strategy to use content length endian

### PR DESCRIPTION
**Pull request by dvargasfuertes**
https://www.loom.com/share/2bdb20cd446042309f91bcb66b1a0ec2

### CODE REVIEW

File: examples/vercel-ai-chatbot/app/api/chat/route.ts [hunk 1]
```diff
@@ -78,8 +78,13 @@ export async function POST(req: Request) {
   const stream = new ReadableStream({
     async pull(controller) {
       const encoder = new TextEncoder()
-      const emit = (event: unknown) =>
-        controller.enqueue(encoder.encode(JSON.stringify(event) + '\n'))
+      const emit = (event: unknown) => {
```
Comment (nihalathanikkal) [inline examples/vercel-ai-chatbot/app/api/chat/route.ts:86]: [nit] Maybe worth a comment here stating a shortened version of what's stated in the video so someone using this as an example knows what's going

File: examples/vercel-ai-chatbot/lib/hooks/use-vellum-chat.tsx [hunk 2]
```diff
@@ -67,85 +142,22 @@ const useVellumChat = ({
...
-          ) as WorkflowResultEventOutputData[]
-
-          parsedChunkValueArray.forEach(parsedChunkValue => {
-            if (
-              parsedChunkValue.state === 'REJECTED' &&
-              parsedChunkValue.type === 'ERROR'
-            ) {
-              throw new Error(parsedChunkValue.value?.message)
+          buffer = concatBuffers(buffer, resultChunk.value)
+          while (buffer.length >= 4) {
```
Comment (nihalathanikkal) [inline examples/vercel-ai-chatbot/lib/hooks/use-vellum-chat.tsx:151]: [nit] Another comment here for why four and about the buffer logic would be helpful

**Review by nihalathanikkal — APPROVED** [review]
Yay! Go AI helping us make solutions for AI 🤖

File: examples/vercel-ai-chatbot/lib/hooks/use-vellum-chat.tsx (hunk 2 above)
Comment (siddseethepalli) [reply examples/vercel-ai-chatbot/lib/hooks/use-vellum-chat.tsx:151]: I think pulling the 4 out into a constant e.g. `UNIT32_SIZE` would make this more clear

**Review by siddseethepalli — APPROVED** [review]

**Pull request merged**